src/isl/isl_type_reference.rs

```diff
@@ -29,7 +50,7 @@ pub enum IslTypeRef {
 impl IslTypeRef {
     /// Creates a [IslTypeRef::CoreIsl] using the [IonType] referenced inside it
     pub fn core(ion_type: IonType) -> IslTypeRef {
```

Comment:
We should reevaluate these method names now that there are two kinds of `core` type.